src/test/UITest.java

```diff
@@ -0,0 +1,101 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class UITest {
+
+    private final ByteArrayOutputStream OUT_CONTENT = new ByteArrayOutputStream();
+    private final PrintStream ORIGINAL_OUT = System.out;
+    private final InputStream ORIGINAL_IN = System.in;
+
+    private final String LINE = " ____________________________________________________________";
+
+    @BeforeEach
+    public void setUpStreams() {
+        System.setOut(new PrintStream(OUT_CONTENT));
+    }
+
+    @AfterEach
+    public void restoreStreams() {
+        System.setOut(ORIGINAL_OUT);
+        System.setIn(ORIGINAL_IN);
+    }
+
+    @Test
+    public void testShowGreeting() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        UI.printGreeting();
+        String expected = "Hello from\n" +
+                logo +
+                "\n" +
+                LINE +
+                "\nHello! I'm Duke\nWhat can I do for you?\n" +
+                LINE +
+                "\n";
```

Comment:
Perhaps it would be better to break before the "+" operator to improve readability?